File: 100-Days-Series/Day 41/MultiplicationOfTwoMatrices.java
class MultiplicationOfTwoMatrices{
     static java.util.Scanner sc=new java.util.Scanner(System.in);
    // Function for multiplication to matrix
    public static void getMatrixMultiplication(int rowsA,int colsA,int colsB){
        int [][]a=new int[rowsA][colsA];
        int [][]b=new int[colsA][colsB];
        int [][]c=new int[rowsA][colsB];
        // Input for Matrix A
        System.out.println("Enter Elements for matrix A ("+rowsA+"x"+colsA+") :");
        for(int i=0;i<rowsA;i++){
            for(int j=0;j<colsA;j++){
                System.out.print("A["+i+"]["+j+"] : ");
                a[i][j]=sc.nextInt();
            }
        }
        // Input for Matrix B
        System.out.println("Enter Elements for matrix B ("+colsA+"x"+colsB+") :");
        for(int i=0;i<rowsA;i++){
            for(int j=0;j<colsA;j++){
                System.out.print("B["+i+"]["+j+"] : ");
                b[i][j]=sc.nextInt();
            }
        }
        // Initializing Matrix C to zero
        for(int i=0;i<rowsA;i++){
            for(int j=0;j<colsB;j++){
                c[i][j]=0;
            }
        }
        // Performing Matrix multiplication
        for(int i=0;i<rowsA;i++){
            for(int j=0;j<colsB;j++){
                for(int k=0;k<colsA;k++){
                    c[i][j]=a[i][k]*b[k][j];
                }
            }
        }
        // Printing the Reusltant matrix
        System.out.println("Resultant Matrix C ("+rowsA+"x"+colsB+") :");
        for(int i=0;i<rowsA;i++){
            for(int j=0;j<colsB;j++){
                System.out.print("C["+i+"]["+j+"] = "+c[i][j]+", ");
            }
            System.out.println();
        }


    }
    // Main Method 
    public static void main(String [] args){
        int rowsA,colsA,colsB;
        System.out.println("----For Matrix A----");
        System.out.print("Enter Rows : ");
        rowsA=sc.nextInt();
        System.out.print("Enter Columns : ");
        colsA=sc.nextInt();
        System.out.print("----For Matrix B----");
        System.out.print("Enter Columns (Rows automatically "+rowsA+") : ");
        colsB=sc.nextInt();
        // Call function to perform multiplication
        getMatrixMultiplication(rowsA, colsA, colsB);
    }
}

File: 100-Days-Series/Day 41/MultiplicationOfTwoMatrices.c
#include<stdio.h>

// Function for matrix multiplication
void matrixMultiplication(int rowsA,int colsA,int colsB){
    int a[rowsA][colsA], b[colsA][colsB], c[rowsA][colsB];
    // Input for matrix A
    printf("\nEnter elements for matrix A (%dx%d):\n",rowsA,colsA);
    for(int i=0;i<rowsA;i++){
        for(int j=0;j<colsA;j++){
            printf("A[%d][%d] : ",i,j);
            scanf("%d",&a[i][j]);
        }
    }
    // Input for matrix B
    printf("Enter element for matrix B (%dx%d):\n",colsA,colsB);
    for(int i=0;i<colsA;i++){
        for(int j=0;j<colsB;j++){
            printf("B[%d][%d]",i,j);
            scanf("%d",&b[i][j]);
        }
    }

    // Initializing Matrix C to zero
    for(int i=0;i<rowsA;i++){
        for(int j=0;j<colsB;j++){
            c[i][j]=0;
        }
    }
    // Performing Matrix Multiplication
    for(int i=0;i<rowsA;i++){
        for(int j=0;j<colsB;j++){
            for(int k=0;k<colsA;k++){
                c[i][j]+=a[i][k]*b[k][j];
            }
        }
    }
    // Printing the Resultant Matrix
    printf("Resultant Matrix C (%dx%d):\n", rowsA, colsB);
    for(int i=0;i<rowsA;i++){
        for(int j=0;j<colsB;j++){
            printf("c[%d][%d]=%d ",i,j,c[i][j]);
        }
        printf("\n");
    }
}

// Let's Start here
void main(){
    int rowsA,colsA,colsB;
    printf("\n----For Matrix A----");
    printf("\nEnter Rows : ");
    scanf("%d",&rowsA);
    printf("\nEnter Columns : ");
    scanf("%d",&colsA);
    printf("\n----For Matrix B----");
    printf("\nEnter Columns (rows automatically %d): ",colsA);
    scanf("%d",&colsB);

    // Call function to perform multiplication
    matrixMultiplication(rowsA,colsA,colsB);

}